qiskit_experiments/visualization/plotters/base_plotter.py

```diff
@@ -0,0 +1,453 @@
+                set to ``None``, and no scaling is applied. If nothing is provided, the axis numbers will
+                be displayed in the scientific notation.
+            yval_unit (str): Unit of y values. See ``xval_unit`` for details.
+            figure_title (str): Title of the figure. Defaults to None, i.e. nothing is shown.
+            series_params (Dict[str, Dict[str, Any]]): A dictionary of plot parameters for each series.
+                This is keyed on the name for each series. Sub-dictionary is expected to have following
+                three configurations, "canvas", "color", and "symbol"; "canvas" is the integer index of
+                axis (when multi-canvas plot is set), "color" is the color of the curve, and "symbol" is
+                the marker style of the curve for scatter plots.
+        """
+        return Options(
+            xlabel=None,
+            ylabel=None,
+            xlim=None,
+            ylim=None,
+            xval_unit=None,
+            yval_unit=None,
+            figure_title=None,
+            series_params={},
+        )
+
+    def set_options(self, **fields):
+        """Set the plotter options.
+
+        Args:
+            fields: The fields to update in options.
+        """
+        self._options.update_options(**fields)
+        self._set_options = self._set_options.union(fields)
+
+    def set_plot_options(self, **fields):
+        """Set the plot options.
+
+        Args:
+            fields: The fields to update in plot options.
+        """
+        self._plot_options.update_options(**fields)
+        self._set_plot_options = self._set_plot_options.union(fields)
+
+    def _initialize_drawer(self):
+        """Configures :attr:`drawer` before plotting.
+
+        The following actions are taken:
+            1. ``axis``, ``subplots``, and ``style`` are passed to :attr:`drawer`.
+            2. ``plot_options`` in :attr:`drawer` are updated based on values set in plotter
+                :attr:`plot_options`
+
+        These steps are different as any plot-option can be passed to :attr:`drawer` if the drawer has a
+        plot-option with the same name. However, ``axis``, ``subplots``, and ``style`` are the only
+        plotter options passed to :attr:`drawer`. This is done because :class:`BasePlotter` distinguishes
+        between plotter options and plot-options.
+        """
+        ## Axis, subplots, and style
+        if self.options.axis:
+            self.drawer.set_options(axis=self.options.axis)
+        if self.options.subplots:
+            self.drawer.set_options(subplots=self.options.subplots)
+        self.drawer.set_plot_options(custom_style=self.options.style)
+
+        ## Plot Options
+        # HACK: We need to accesses internal variables of the Options class, which is not good practice.
+        # Options._fields are dictionaries. However, given we are accessing an internal variable, this
+        # may change in the future.
```

Comment:
This is not super nice but I also cannot think of a better way to do this given the way that `Options` is built.

Reply:
I've updated `PlotStyle` to have a `__dict__` property. There was some confusion I had with how `Options` defines `__dict__` and `_fields`, given that it has some Python magic to achieve the functionality it requires. Looking back at its implementation allowed me to properly implement a `__dict__` property for `PlotStyle. See 9f4e1ee for the changes and additional tests.